addons/stock/models/stock_move.py

```diff
@@ -1287,19 +1238,6 @@ def _onchange_lot_ids(self):
                 'warning': {'title': _('Warning'), 'message': _('Unavailable Serial numbers. Please correct the serial numbers encoded: %(serial_numbers_to_locations)s', serial_numbers_to_locations=sn_to_location)}
             }
 
-    @api.onchange('product_uom')
-    def _onchange_product_uom(self):
-        if self.product_uom.factor > self.product_id.uom_id.factor:
```

Comment:
I think this warning is still relevant, although we'd need to scrap/change the part about changing the rounding precision.